### PR TITLE
Update group EP

### PR DESCRIPTION
**Pull request by jarrydlee**
fixes https://github.com/PagerNation/PagerNation/issues/105

### CODE REVIEW

File: server/routes/group.js [hunk 1]
```diff
@@ -19,6 +19,9 @@ router.route('/:groupName/user')
 router.route('/:groupName/user/:userId')
   .delete(validate(paramValidation.userById), groupCtrl.removeUser);
 
+router.route('/:groupName/user/:userId/escalationpolicy')
+  .post(groupCtrl.updateEscalationPolicy);
```
Comment (jarrydlee) [inline server/routes/group.js]: Validation needed

**Comment by jarrydlee** [comment]
@kalebdavis can you give this a squiz?

File: server/services/group.js [hunk 2]
```diff
@@ -14,7 +15,8 @@ function deleteGroup(groupName) {
 function createGroup(groupObject) {
   const groupSchema = Joi.object().keys({
     name: Joi.string().required(),
-    users: Joi.array().items(Joi.string().hex().length(24))
+    users: Joi.array().items(Joi.string().hex().length(24)),
+    escalationPolicy: Joi.object()
```
Comment (jarrydlee) [inline server/services/group.js]: I think this should be changed. The EP should be automatically created with the group and updated after creation if needed. 

I will do this in a separate PR so this doesn't get too cluttered. See https://github.com/PagerNation/PagerNation/issues/118

File: server/tests/services/group.test.js [hunk 3]
```diff
@@ -129,7 +129,7 @@ describe('# Group Service', () => {
       });
     });
 
-    describe('# addUser', () => {
+    describe('# addUser()', () => {
```
Comment (kalebdavis) [inline server/tests/services/group.test.js]: Cleanup, nice

File: server/tests/services/group.test.js [hunk 4]
```diff
@@ -197,5 +197,51 @@ describe('# Group Service', () => {
...
+        build('escalationPolicy', fixtures.escalationPolicy({ subscribers: [userId] }))
+          .then(escalationPolicy => build('group', fixtures.group({ escalationPolicy: escalationPolicy })))
+          .then((newGroup) => {
+            group = newGroup;
+            done();
+          });
+      });
+
+
+      it('should update a groups escalation policy', (done) => {
```
Comment (kalebdavis) [inline server/tests/services/group.test.js]: 100% a nitpick, but can you reformat so the grammar here is correct? (groups => group's)

Comment (kalebdavis) [inline server/tests/services/group.test.js]: Also, these should be wrapped in contexts.

`context('with incorrect fields')`, etc

File: server/routes/group.js [hunk 5]
```diff
@@ -19,6 +19,9 @@ router.route('/:groupName/user')
 router.route('/:groupName/user/:userId')
   .delete(validate(paramValidation.userById), groupCtrl.removeUser);
 
+router.route('/:groupName/user/:userId/escalationpolicy')
+  .put(validate(paramValidation.updateEscalationPolicy), groupCtrl.updateEscalationPolicy);
```
Comment (jarrydlee) [inline server/routes/group.js]: Yep, I confused this for the PR you're doing

File: server/tests/services/group.test.js [hunk 6]
```diff
@@ -212,35 +212,38 @@ describe('# Group Service', () => {
...
-            expect(savedGroup.escalationPolicy.rotationInterval)
-              .to.equal(newEscalationPolicy.rotationInterval);
-            done();
-          });
+      context('with correct body fields', () => {
+        it('should update a group\'s escalation policy', (done) => {
+          const newEscalationPolicy = {
+            pagingInterval: 1111,
+            rotationInterval: 2222
+          };
```
Comment (kalebdavis) [inline server/tests/services/group.test.js]: This setup should be pulled out of the `it` block

    context('with correct body fields) 
      const newEscalationPolicy = {}
      it('should...

**Pull request merged**